src/main/resources/avro/referencemethods.avdl

```diff
@@ -325,4 +396,99 @@ GetSequenceBasesResponse getSequenceBases(
     /** Additional request parameters to restrict the query. */
     GetSequenceBasesRequest request) throws GAException;
 
+/****************  /joins/search  *******************/
+/**
+This request maps to the body of `POST /joins/search`
+as JSON.
+*/
+record SearchJoinsRequest {
+  /**
+  If present, return only `Join`s which belong to this reference set.
+  */
+  union { null, string } referenceSetId = null;
+  
+  /**
+  If present, return only `Join`s which belong to this variant set.
+  */
+  union { null, string } variantSetId = null;
+
+  // TODO: Specify a proper sort of filter object instead of this. We want to be
+  // able to support a list of sequences, but also a single sequence and a
+  // range.
+
+  /**
+  If nonempty, return only `Join`s with at least one `Position` on one of these
+  sequences.
+  */
+  array<string> sequenceIds = [];
+  
+  /**
+  Must not contain more entries than `sequenceIds`.
```

Comment:
I agree that this model is vague; I think @adamnovak's comment confirms this ;). However, as it is consistent with _other_ GA4GH APIs, I don't think we should hold up the PR on this. As an aside, there's no reason that the query definition needs to be unstructured in order to allow implementations to implement a flexible execution model. E.g., databases rewrite user provided queries in order to improve query performance. I think it would make sense to refactor the way that we express queries, but that's a longer debate and not really germane to this discussion.